### PR TITLE
vsock: Add iommu test for vsock

### PR DESCRIPTION
**Pull request by zhenyzha**
Add iommu test for vsock

ID: 1961987
Signed-off-by: Zhenyu Zhang <zhenyzha@redhat.com>

### CODE REVIEW

**Comment by zhenyzha** [comment]
git am 2829.patch 
Applying: vsock: Add iommu test for vsock

` (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.iommu_enabled.arm64-pci: PASS (115.55 s)`
@PaulYuuu @qcheng-redhat Hello Could you pls help review, thanks.

File: qemu/tests/cfg/vsock.cfg [hunk 1]
```diff
@@ -24,3 +24,11 @@
               devno = 'fe.1.2345'
         - migration_with_vsock:
             type = migration_with_vsock
+        - iommu_enabled:
+            no RHEL.6 RHEL.7 RHEL.8.0 RHEL.8.1 RHEL.8.2 RHEL.8.3 RHEL.8.4
+            required_qemu = [5.2.0,)
+            #refer to bug 1879052,1894101
+            virtio_dev_iommu_platform = on
+            ppc64, ppc64le:
```
Comment (PaulYuuu) [inline qemu/tests/cfg/vsock.cfg:32]: `pc` machine should also apply these two parameters if I understand correctly.

Comment (zhenyzha) [reply qemu/tests/cfg/vsock.cfg:32]: @qcheng-redhat hello, I checked the existing tp-qemu case and found that iommu tests are only on q35.
So this test is for x86: `only q35` or `pc,  ppc64, ppc64le:`
Could you confirm this?

Comment (qcheng-redhat) [reply qemu/tests/cfg/vsock.cfg:32]: For x86, only q35.

Comment (zhenyzha) [reply qemu/tests/cfg/vsock.cfg:34]: thanks all, Corrected.

**Comment by zhenyzha** [comment]
git am 2829.patch 
Applying: vsock: Add iommu test for vsock

` (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.iommu_enabled.arm64-pci: PASS (117.50 s)`

@PaulYuuu @qcheng-redhat Hello Could you pls help review again, thanks.

File: qemu/tests/cfg/vsock.cfg [hunk 2]
```diff
@@ -24,3 +24,13 @@
               devno = 'fe.1.2345'
         - migration_with_vsock:
             type = migration_with_vsock
+        - iommu_enabled:
+            no RHEL.6 RHEL.7 RHEL.8.0 RHEL.8.1 RHEL.8.2 RHEL.8.3 RHEL.8.4
+            required_qemu = [5.2.0,)
+            #refer to bug 1879052,1894101
+            x86_64:
```
Comment (PaulYuuu) [inline qemu/tests/cfg/vsock.cfg:31]:
```
machine_type_extra_params = "kernel-irqchip=split"
HostCpuVendor.intel:
    intel_iommu = yes
    enable_guest_iommu = yes
```
These lines may be needed.

Comment (zhenyzha) [reply qemu/tests/cfg/vsock.cfg:31]: Thanks for the reminder @PaulYuuu 
@qcheng-redhat hello, about this test, do we need to add the following parameters on x86?

```
machine_type_extra_params = "kernel-irqchip=split"
HostCpuVendor.intel:
    intel_iommu = yes
    enable_guest_iommu = yes
virtio_dev_ats = on
```

Comment (qcheng-redhat) [reply qemu/tests/cfg/vsock.cfg:31]: Hi @zhenyzha I think these parameters are for intel_iommu, as we talked in the chat.

Comment (zhenyzha) [reply qemu/tests/cfg/vsock.cfg:32]: Ok, corrected.

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.vsock_test.iommu_enabled.q35: PASS (50.43 s)

@PaulYuuu @qcheng-redhat Could you pls help review again when you are free. thanks.

**Comment by qcheng-redhat** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.vsock_test.iommu_enabled.q35: PASS (49.82 s)

LGTM

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**